src/ser.rs

```diff
@@ -90,30 +98,50 @@ impl<'py> Serialize for SerializePyObject<'py> {
             for (k, v) in simple_items {
                 let key = table_key(k)?;
                 let value = self.with_obj(v);
+                let v_ob_type = v.get_type_ptr() as usize;
+                if v_ob_type == lookup.none && self.omit_none {
+                    continue;
+                }
                 map.serialize_entry(key, &value)?;
             }
             for (k, v) in array_items {
                 let key = table_key(k)?;
                 let value = self.with_obj(v);
+                let v_ob_type = v.get_type_ptr() as usize;
+                if v_ob_type == lookup.none && self.omit_none {
+                    continue;
+                }
                 map.serialize_entry(key, &value)?;
             }
             for (k, v) in dict_items {
                 let key = table_key(k)?;
                 let value = self.with_obj(v);
+                let v_ob_type = v.get_type_ptr() as usize;
+                if v_ob_type == lookup.none && self.omit_none {
+                    continue;
+                }
                 map.serialize_entry(key, &value)?;
             }
             map.end()
         } else if ob_type == lookup.list {
             let py_list: &PyList = self.obj.cast_as().map_err(map_py_err)?;
             let mut seq = serializer.serialize_seq(Some(py_list.len()))?;
             for element in py_list {
+                let v_ob_type = element.get_type_ptr() as usize;
+                if v_ob_type == lookup.none && self.omit_none {
+                    continue;
+                }
                 seq.serialize_element(&self.with_obj(element))?
             }
             seq.end()
         } else if ob_type == lookup.tuple {
             let py_tuple: &PyTuple = self.obj.cast_as().map_err(map_py_err)?;
             let mut seq = serializer.serialize_seq(Some(py_tuple.len()))?;
             for element in py_tuple {
+                let v_ob_type = element.get_type_ptr() as usize;
+                if v_ob_type == lookup.none && self.omit_none {
+                    continue;
+                }
                 seq.serialize_element(&self.with_obj(element))?
```

Comment:
```suggestion
                if self.omit_none && element.is_none() {
                    continue;
                }
                seq.serialize_element(&self.with_obj(element))?
```

Since we're only looking up one type, `.is_none()` will probably be faster than `element.get_type_ptr() as usize == lookup.none`.